core/pkg/artifacts/saver.go

```diff
@@ -138,6 +140,41 @@ func (as *ArtifactSaver) createManifest(
 	return response.GetCreateArtifactManifest().ArtifactManifest, nil
 }
 
+func (as *ArtifactSaver) updateManifest(
+	artifactManifestId string, manifestDigest string,
+) (attrs updateArtifactManifestAttrs, rerr error) {
+	response, err := gql.UpdateArtifactManifest(
+		as.Ctx,
+		as.GraphqlClient,
+		artifactManifestId,
+		&manifestDigest,
+		nil,
+		true,
+	)
+	if err != nil {
+		return updateArtifactManifestAttrs{}, err
+	}
+	return response.GetUpdateArtifactManifest().ArtifactManifest, nil
```

Comment:
Could the UpdateArtifactMenifest be nil?

should we check nil response?

Reply:
I don't think this should ever be nil, unless there was an error

Reply:
right and in that case you would want to return an error?

Reply:
yes that should be done by the check before this, right? or am I missing something here?
```
	if err != nil {
		return updateArtifactManifestAttrs{}, err
	}
```

Reply:
i was just try to be extra careful and check that the response parts are also valid.

```
response.GetUpdateArtifactManifest().ArtifactManifest
```

- if response is nil: `GetUpdateArtifactManifest()` will fail
- and same for if `UpdateArtifactManifest` nil again that accessing `ArtifactManifest` will fail

Reply:
makes sense, I can add an extra check just to be safe